modules/react/toast/lib/Toast.tsx

```diff
@@ -1,118 +1,107 @@
+ *    </Toast>
+ * );
+ *```
+ */
+export const Toast = createContainer('div')({
   displayName: 'Toast',
-  Component: (
-    {
-      icon = checkIcon, // needed for TS2742 - https://github.com/microsoft/TypeScript/issues/29808
-      iconColor = colors.greenApple400,
-      onClose,
-      onActionClick,
-      actionText,
-      children,
-      ...elemProps
-    }: ToastProps,
-    ref,
-    Element
-  ) => {
-    const isInteractive = onClose || onActionClick;
-    const isError = iconColor === colors.cinnamon500 && icon === exclamationCircleIcon;
-
-    return (
-      <Popup.Card
-        ref={ref}
-        as={Element}
-        width={toastWidth}
-        depth={5}
-        role={isInteractive ? 'dialog' : isError ? 'alert' : 'status'}
-        aria-live={isInteractive ? 'off' : isError ? 'assertive' : 'polite'}
-        aria-atomic={!isInteractive}
-        padding="xxs"
-        {...elemProps}
-      >
-        {onClose && <Popup.CloseIcon aria-label="Close" onClick={onClose} size="small" />}
-        <Popup.Body>
-          <ToastContentContainer onClose={onClose}>
-            {icon && <ToastSystemIcon color={iconColor} colorHover={iconColor} icon={icon} />}
-            <Message>
-              {children}
-              {onActionClick && (
-                <StyledActionButton as="button" onClick={onActionClick}>
-                  {actionText}
-                </StyledActionButton>
-              )}
-            </Message>
-          </ToastContentContainer>
-        </Popup.Body>
-      </Popup.Card>
-    );
+  modelHook: useToastModel,
+  subComponents: {
+    /**
+     * `Toast.Body` should container `Toast.Message` and `Toast.Link`. This ensures proper styling and spacing between elements.
+     *
+     * ```tsx
+     * <Toast.Body>
+     *  <Toast.Message>Your workbook was successfully processed.</Toast.Message>
+     *  <Toast.Link href="#hreflink">Custom Link</Toast.Link>
+     * </Toast.Body>
+     * ```
+     */
+    Body: ToastBody,
+    /**
+     * `Toast.CloseIcon` renders a {@link PopupCloseIcon}. You can pass an `onClick` when used with `Popper` to dismiss the `Toast`.
+     */
+    CloseIcon: ToastCloseIcon,
+    /**
+     * `ToastIcon` renders a `SystemIcon` where you have access to all styling properties from `SystemIcon`.
+     */
+    Icon: ToastIcon,
+    /**
+     * `Toast.Message` renders our `Subtext` component where you can style however way you'd like with style properties.
+     * This component also has an `id` so that when the `Toast` has a prop of `mode="dialog"` the message is read out by screen readers.
+     */
+    Message: ToastMessage,
+    /**
+     * `Toast.Link` renders our `Hyperlink` component. If you need to link to more information, use this component.
+     */
+    Link: ToastLink,
   },
+})<ToastProps>(({children, ...elemProps}, Element, model) => {
+  const getAriaAttributes = (mode: string): React.HtmlHTMLAttributes<HTMLDivElement> => {
```

Comment:
What do you think to have a `getAriaAttributes` function outside component scope, so component will be more visible?